builder/azure/arm/config.go

```diff
@@ -196,6 +197,9 @@ func (c *Config) toImageParameters() *compute.Image {
 			SourceVirtualMachine: &compute.SubResource{
 				ID: to.StringPtr(c.toVMID()),
 			},
+			StorageProfile: &compute.ImageStorageProfile{
```

Comment:
Change looks good to me minus the lack of unit tests. Please add two tests. One test to assert the default value, and one test to assert the value is changed as expected.